enhancements/local-storage/automatic-detection.md

```diff
@@ -0,0 +1,264 @@
+    MinDeviceCount int `json:"minDeviceCount"`
+    // +optional
+    // Maximum number of Devices that needs to be detected per node.
+    MaxDeviceCount int `json:"maxDeviceCount"`
+    // VolumeMode determines whether the PV created is Block or Filesystem
+    // + optional
+    VolumeMode PersistentVolumeMode `json:"volumeMode,omitempty"`
+    // FSType type to be used when volumeMode is Filesystem
+    // +optional
+    FSType string `json:"fsType,omitempty"`
+    // DeviceInclusionSpec are the filtration rules for including a device in the device discovery
+    // +optional
+    DeviceInclusionSpec *DeviceInclusionSpec `json:"deviceInclusionSpec"`
+}
+
+// DeviceMechanicalProperty holds the device's mechanical spec. It can be rotational, nonRotational or
+// RotationalAndNonRotational
+type DeviceMechanicalProperty string
+
+const (
+    // The mechanical properties of the devices
+    // Rotational refers to magnetic disks
+    Rotational DeviceMechanicalProperty = "rotational"
+    // NonRotational refers to ssds
+    NonRotational DeviceMechanicalProperty = "nonRotational"
+    // RotationalAndNonRotational refers to both magnetic and ssds
+    RotationalAndNonRotational DeviceMechanicalProperty = "rotationalAndNonRotational"
+)
+
+type DeviceDiscoveryPolicyType string
+
+const (
+    // The DeviceDiscoveryPolicies that will be supported by the LSO.
+    // These Discovery policies will based on lsblk's type output
+    // Disk represents a device-type of disk
+    Disk DeviceDiscoveryPolicyType = "disk"
+    // Part represents a device-type of partion
+    Part DeviceDiscoveryPolicyType = "part"
+)
+
+type DeviceInclusionSpec struct {
+    // DeviceTypes that should be used for automatic detection. This would be one of the types supported
+    // by the local-storage operator. Currently the supported types are: disk,part
+    // If the list is empty the default value will be `[disk]`.
+    DeviceTypes []DeviceDiscoveryPolicyType `json:"deviceType"`
+
+    // DeviceMechanicalProperty denotes whether Rotational or NonRotational disks should be used.
+    // by default, RotationalAndNonRotational which matches all disks.
+    // +optional
+    DeviceMechanicalProperty DeviceMechanicalProperty `json:"deviceMechanicalProperty"`
+
+    // MinSize is the minimum size of the device which needs to be included
+    // +optional
+    MinSize resource.Quantity `json:"minSize"`
+
+    // MaxSize is the maximum size of the device which needs to be included
+    // +optional
+    MaxSize resource.Quantity `json:"maxSize"`
+
+    // Models is a list of device models. If not empty, the device's model as outputted by lsblk needs
+    // to contain at least one of these strings.
+    // +optional
+    Models []string `json:"models"`
+    // Vendors is a list of device vendors. If not empty, the device's model as outputted by lsblk needs
+    // to contain at least one of these strings.
+    // +optional
+    Vendors []string `json:"vendors"`
```

Comment:
Do we expect to use these in the first cut the API? As discussed in the call - it is easy to think of a field as "nice to have", but we will not be able to remove these fields from the API once added and we will have to forever support quirks associated with model and vendor names.

Reply:
These were suggested by @leseb as a good to have. @leseb can we have this in future implementation. I can open an issue to keep this on track?

Reply:
I can't think of a good reason not to have this in the first iteration. So I'd vote to keep it.

Reply:
cool removed

Reply:
okay

Reply:
I would still argue that these can be added later if needed. How likely it is that customer is adding disks from different vendors/models and wants to create storageclasses sorted by vendors/models?  Some of the fields in LSO api was added as customers asked them, such fixes can be backported if they are backward compatible.

Reply:
If you don't have these fields and you have NVMe and SSD drives of the same size then you cannot tell the difference.

Reply:
can we have NVMe as a new discover policy ?

Reply:
Why? How?

Reply:
https://github.com/openshift/enhancements/pull/190/files#diff-ea6b51c2a2234cd3fb1280d7332fd70aR141 here adding one more constant as nvme

Reply:
The only way to detect an NVMe is via the `TRAN` output of lsblk, the `TYPE` won't work.

Reply:
While I agree we need to be careful about every property we add, model and vendor seem very useful, even if not frequently needed.

Reply:
cool keeping this for now.